app.js

```diff
@@ -0,0 +1,69 @@
+const getVagonQuantity = (max = 20, min = 10) =>
+  Math.round(min - 0.5 + Math.random() * (max - min + 1));
+
+class Train {
+  constructor(sizeMax) {
+		this.vagons = [],
+		this.vagonCounter = 0;
+    // генерим выгоны
+    for (var i = 0; i < getVagonQuantity(sizeMax); i++) {
+      this.vagons.push({
+        lightIsOn: !!Math.round(Math.random()),
+      });
+    }
+  }
+  get forwardStep() {
+    // замыкаем вагоны
+    return this.vagons[++this.vagonCounter % this.vagons.length];
```

Comment:
you start from the first vagon and calculate according to train length - this is not correct.

Reply:
Then I have to do the meter elsewhere.